browser/src/core/Socket.js

```diff
@@ -289,7 +289,28 @@ app.definitions.Socket = L.Class.extend({
 									       {'_slurpQueue.length' : String(queueLength)});
 		if (this._map && this._map._docLayer) {
 			this._map._docLayer.pauseDrawing();
+
+			// Queue an instant timeout early to try to measure the
+			// re-rendering delay before we get back to the main-loop.
+			if (this.traceEventRecordingToggle)
+			{
```

Comment:
Why don't you use the existing createCompleteTraceEvent function (in browser/src/core/Socket.js) but duplicate code from it?